content/en/docs/installing-on-linux/on-premises/install-kubesphere-on-vmware-vsphere.md

```diff
@@ -0,0 +1,401 @@
+# main frontend which proxys to the backends
+#---------------------------------------------------------------------
+frontend  kube-apiserver
+    bind *:6443
+    mode tcp
+    option tcplog
+    default_backend kube-apiserver
+#---------------------------------------------------------------------
+# round robin balancing between the various backends
+#---------------------------------------------------------------------
+backend kube-apiserver
+    mode tcp
+    option tcplog
+    balance     roundrobin
+    default-server inter 10s downinter 5s rise 2 fall 2 slowstart 60s maxconn 250 maxqueue 256 weight 100
+    server kube-apiserver-1 10.10.71.214:6443 check
+    server kube-apiserver-2 10.10.71.73:6443 check
+    server kube-apiserver-3 10.10.71.62:6443 check
+```
+```bash
+#Check for grammar before starting
+haproxy -f /etc/haproxy/haproxy.cfg -c
+#Start Haproxy and set it to enable haproxy
+systemctl restart haproxy && systemctl enable haproxy
+#Stop Haproxy
+systemctl stop haproxy
+```
+### 3. Configure Keepalived
+```bash
+# Main haproxy 77 lb-0-10.10.71.77
+#/etc/keepalived/keepalived.conf
+global_defs {
+  notification_email {
+  }
+  smtp_connect_timeout 30   
+  router_id LVS_DEVEL01 
+  vrrp_skip_check_adv_addr
+  vrrp_garp_interval 0
+  vrrp_gna_interval 0
+}
+vrrp_script chk_haproxy {
+  script "killall -0 haproxy"
+  interval 2
+  weight 2
+}
+vrrp_instance haproxy-vip {
+  state MASTER  
+  priority 100  
+  interface ens192                       
+  virtual_router_id 60 
+  advert_int 1 
+  authentication {
+    auth_type PASS 
+    auth_pass 1111
+  }
+  unicast_src_ip 10.10.71.77     
+  unicast_peer {
+    10.10.71.66                      
+  }
+  virtual_ipaddress {
+    #vip
+    10.10.71.67/24 
+  }
+  track_script {
+    chk_haproxy
+  }
+}
+```
+```bash
+#remarks haproxy 66 lb-1-10.10.71.66
+#/etc/keepalived/keepalived.conf
+global_defs {
+  notification_email {
+  }
+  router_id LVS_DEVEL02 
+  vrrp_skip_check_adv_addr
+  vrrp_garp_interval 0
+  vrrp_gna_interval 0
+}
+vrrp_script chk_haproxy {
+  script "killall -0 haproxy"
+  interval 2
+  weight 2
+}
+vrrp_instance haproxy-vip {
+  state BACKUP 
+  priority 90
+  interface ens192                        
+  virtual_router_id 60
+  advert_int 1
+  authentication {
+    auth_type PASS
+    auth_pass 1111
+  }
+  unicast_src_ip 10.10.71.66      
+  unicast_peer {
+    10.10.71.77                        
+  }
+  virtual_ipaddress {
+    10.10.71.67/24 
+  }
+  track_script {
+    chk_haproxy
+  }
+}
+```
+```bash
+#start keepalived and enable keepalived
+systemctl restart keepalived && systemctl enable keepalived
+systemctl stop keepaliv
+systemctl start keepalived    
+```
+
+### 4. Verify availability
+
+- Use `ip a s` to view the vip binding status of each lb node
+- Pause VIP node haproxy：`systemctl stop haproxy`
+- Use `ip a s` again to check the vip binding of each lb node, and check whether vip drifts
+- Or use `systemctl status -l keepalived` command to view
```

Comment:
This section looks not that clear, we are not sure about the correct return and its explanation.